AngelaGuardia/Crate
# Update subscription mutation

#### What's this PR do?

- This PR implements the `subscriptionUpdate` mutation to enable editing of the `nextDeliveryDate` attribute.

#### Where should the reviewer start?

- Start within the `./code/api/src/modules/subscription` directory. First, we modified the `model.js` file to include `nextDeliveryDate`. Then, we updated the `types.js` file to include `nextDeliveryDate` as well. We built out the mutation within the `mutations.js` file to only allow `nextDeliveryDate` to be updated. Since subscriptions operate as a joins table between users and crates within this project, we didn't add `userID` or `crateID` to this update mutation because each of these operate as our foreign keys. 

#### How should this be manually tested?

- Within the `./code/api` directory, run `npm start`. Open a browser window and navigate to localhost:8000 to enter the GraphiQL interface. In the upper right hand corner, click on Docs, then click on mutations (the yellow link) to view all of the mutations we have setup for this project. Click on subscriptionUpdate to see the fields.

#### Any background context you want to provide?

- Within our database, we are saving `nextDeliveryDate` as a DATE type to the subscription model. GraphiQL's output, however, returns a string with the date as an integer. For example, if we have a date within our database like `"2020-01-02 15:47:01.537-07"`, the output is a string like `"1609627621537".`

#### What are the relevant tickets?
- Closes #27 

#### Screenshots (if appropriate)
- ![subscriptionUpdate-graphiql](https://user-images.githubusercontent.com/57038617/101701114-1c736600-3a33-11eb-8d28-7639ff5d18af.png)

#### Questions:
#### New Feature Submissions:
 * Have you added an explanation of what your changes do? 👍🏽
 * Have you linted your code locally prior to submission? 👍🏽
#### Testing:
 * Have you written new tests for your core changes, as applicable? Not Yet
 * Have you successfully ran tests with your changes locally? Confirmed with GraphiQL
